src/Components/Blazor/Blazor/src/Hosting/WebAssemblyHost.cs

```diff
@@ -19,6 +19,11 @@ internal class WebAssemblyHost : IWebAssemblyHost
 
         public WebAssemblyHost(IServiceProvider services, IJSRuntime runtime)
         {
+            // To ensure JS-invoked methods don't get linked out, have a reference to their enclosing types
+            GC.KeepAlive(typeof(EntrypointInvoker));
+            GC.KeepAlive(typeof(JSInteropMethods));
+            GC.KeepAlive(typeof(WebAssemblyEventDispatcher));
```

Comment:
FYI there's an attribute called `[PreserveDependency]` that has this effect. I'm not sure about the status of it right now, so I'm not suggesting you change the code, but this kind of usage is getting thought about.

Reply:
I'm aware of it, but didn't want to use it because you have to specify the dependency as a string in a particular format. I preferred to declare the dependency in a strongly-typed way so it will survive future refactoring.

Also you have to create your own `PreserveDependencyAttribute` which would be OK if we were using it a lot in this assembly, but it feels like a bunch of extra noise for something handled more conveniently via these three lines.